### PR TITLE
fix(Hotfix/soehlert/username): Fill out API username as well as WUI username

### PR DESCRIPTION
**Pull request by samoehlert**
We were silently dropping the username from the json data in API calls. This meant records coming from automated tools (eg Zeek) were not attributed to anyone. This fixes that while still automatically reading the logged in user from the WUI if an entry is added that way.

Also had to update tests to use the API correctly.

### CODE REVIEW

File: scram/route_manager/api/serializers.py [hunk 1]
```diff
@@ -38,7 +38,11 @@ class EntrySerializer(serializers.HyperlinkedModelSerializer):
     )
     route = rest_framework.CidrAddressField()
     actiontype = serializers.CharField(default="block")
-    who = CurrentUserDefault()
```
Comment (grigorescu) [inline scram/route_manager/api/serializers.py:41]: Would it be more or less confusing if this just read:

`who = CurrentUserDefault() or serializes.CharField()`?

**Review by grigorescu — APPROVED** [review]
Left one style suggestion (can't tell if it's better or worse), in 2 places. Looks good to me either way.

File: scram/route_manager/api/serializers.py (hunk 1 above)
Comment (samoehlert) [reply scram/route_manager/api/serializers.py:41]: After looking into this, I find the proposed syntax more confusing (and asked around a little bit to the same response) so for the sake of future us, I'm going to leave the more obvious syntax.

File: scram/route_manager/api/views.py [hunk 2]
```diff
@@ -64,7 +64,12 @@ def get_permissions(self):
     def perform_create(self, serializer):
         actiontype = serializer.validated_data["actiontype"]
         route = serializer.validated_data["route"]
-
+        if self.request.user.username:
```
[the remark this answers is not in the document]
Comment (samoehlert) [reply scram/route_manager/api/views.py:67]: Ditto my answer from above

**Pull request merged**